File: library/src/test/java/com/proxerme/library/api/EnumRetrofitConverterFactoryTest.java
package com.proxerme.library.api;

import com.proxerme.library.entitiy.info.Entry;
import com.proxerme.library.enums.Genre;
import org.junit.Before;
import org.junit.Test;
import retrofit2.Converter;

import java.io.IOException;
import java.lang.annotation.Annotation;

import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;

/**
 * @author Ruben Gees
 */
public class EnumRetrofitConverterFactoryTest {

    private EnumRetrofitConverterFactory factory;

    @Before
    public void setUp() {
        factory = new EnumRetrofitConverterFactory();
    }

    @Test
    public void testCreate() {
        assertThat(factory.stringConverter(Genre.class, new Annotation[0], null)).isNotNull();
    }

    @Test
    public void testCreateNoEnum() {
        assertThat(factory.stringConverter(Entry.class, new Annotation[0], null)).isNull();
    }

    @SuppressWarnings({"unchecked", "ConstantConditions"})
    @Test
    public void testConvert() throws IOException {
        assertThat(((Converter<Enum<?>, String>) factory.stringConverter(Genre.class, new Annotation[0], null))
                .convert(Genre.ADVENTURE)).isEqualTo("Abenteuer");
    }
}

File: library/src/test/kotlin/me/proxer/library/internal/adapter/EnumRetrofitConverterFactoryTest.kt
package me.proxer.library.internal.adapter

import me.proxer.library.entity.info.Entry
import me.proxer.library.enums.MediaLanguage
import org.amshove.kluent.shouldBeNull
import org.amshove.kluent.shouldEqual
import org.amshove.kluent.shouldNotBeNull
import org.junit.jupiter.api.Test
import retrofit2.Retrofit

/**
 * @author Ruben Gees
 */
class EnumRetrofitConverterFactoryTest {

    private val factory = EnumRetrofitConverterFactory()
    private val retrofit = Retrofit.Builder().baseUrl("https://proxer.me").build()

    @Test
    fun testCreate() {
        factory.stringConverter(MediaLanguage::class.java, emptyArray(), retrofit).shouldNotBeNull()
    }

    @Test
    fun testCreateNoEnum() {
        factory.stringConverter(Entry::class.java, emptyArray(), retrofit).shouldBeNull()
    }

    @Test
    fun testConvert() {
        val converter = factory.stringConverter(MediaLanguage::class.java, emptyArray(), retrofit)

        converter.shouldNotBeNull()
        converter.convert(MediaLanguage.GERMAN) shouldEqual "de"
    }
}
